### PR TITLE
refactor(core): Update dynamic node parameter endpoints to use DTOs

### PR DESCRIPTION
**Pull request by netroy**
## Summary

This is part 4 in a series of refactors.
Previous PRs:
- https://github.com/n8n-io/n8n/pull/12373
- https://github.com/n8n-io/n8n/pull/12375
- https://github.com/n8n-io/n8n/pull/12377

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [x] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

**Review by RicardoE105 — DISMISSED** [review]

File: packages/cli/src/controllers/__tests__/dynamic-node-parameters.controller.test.ts [hunk 1]
```diff
@@ -1,34 +1,223 @@
...
+
+		it('should call getOptionsViaMethodName when methodName is provided', async () => {
+			const payload: OptionsRequestDto = {
+				...basePayload,
+				methodName: 'testMethod',
+			};
+			const req = { user: mockUser } as AuthenticatedRequest;
+
+			const expectedResult: INodePropertyOptions[] = [{ name: 'test', value: 'value' }];
+			service.getOptionsViaMethodName.mockResolvedValue(expectedResult);
```
[the remark this answers is not in the document]
Comment (netroy) [reply packages/cli/src/controllers/__tests__/dynamic-node-parameters.controller.test.ts:51]: that's definitely a problem. quite a lot of our integration tests are using mocks. I tried fixing it here, but it very quickly became a side quest of its own. I'll address this in a separate PR.

**Review by RicardoE105 — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.74.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.74.1)

**Comment by janober** [comment]
Got released with [`n8n@1.74.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.74.1)

**Comment by janober** [comment]
Got released with [`n8n@1.74.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.74.0)

**Comment by janober** [comment]
Got released with [`n8n@1.74.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.74.0)

**Pull request merged**